crates/wasi/src/p3/bindings.rs

```diff
@@ -0,0 +1,306 @@
+//!             "wasi:clocks/monotonic-clock@0.3.0#[async]wait-until",
+//!             "wasi:filesystem/types@0.3.0#[method]descriptor.read-via-stream",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.write-via-stream",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.append-via-stream",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.advise",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.sync-data",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.get-flags",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.get-type",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.set-size",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.set-times",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.read-directory",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.sync",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.create-directory-at",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.stat",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.stat-at",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.set-times-at",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.link-at",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.open-at",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.readlink-at",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.remove-directory-at",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.rename-at",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.symlink-at",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.unlink-file-at",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.is-same-object",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.metadata-hash",
+//!             "wasi:filesystem/types@0.3.0#[async method]descriptor.metadata-hash-at",
+//!             "wasi:sockets/ip-name-lookup@0.3.0#[async]resolve-addresses",
+//!             "wasi:sockets/types@0.3.0#[async method]tcp-socket.connect",
+//!             "wasi:sockets/types@0.3.0#[async method]tcp-socket.send",
+//!             "wasi:sockets/types@0.3.0#[async method]udp-socket.receive",
+//!             "wasi:sockets/types@0.3.0#[async method]udp-socket.send",
+//!             "wasi:sockets/types@0.3.0#[method]tcp-socket.bind",
+//!             "wasi:sockets/types@0.3.0#[method]tcp-socket.listen",
+//!             "wasi:sockets/types@0.3.0#[method]tcp-socket.receive",
+//!             "wasi:sockets/types@0.3.0#[method]udp-socket.bind",
+//!             "wasi:sockets/types@0.3.0#[method]udp-socket.connect",
+//!         ],
+//!     },
+//! });
```

Comment:
Ideally we're going to want to smith this if we can, the invocation here is quite large and I'd want to dig in to why the `async: { ... }` bits need to be specified (if they even still do, unsure)

Reply:
Documented in #11246 and added a TODO with a ref